csrc/host_ir/jit.cpp

```diff
@@ -1258,34 +1275,223 @@ void HostIrJitImpl::registerExternalFunctions() {
+          input_bytes +=
+              static_cast<int64_t>(input_tensors[i]->storage().nbytes());
         }
-        for (int64_t i = 0; i < num_outputs; i++) {
-          output_args.push(*output_tensors[i]);
+      }
+      sprof.inputBytesAccessed(input_bytes);
+
+      // Set scheduler type from compiled kernel
+      CompiledKernel* ck = launch_kernel_ptr->compiledKernel();
+      if (ck != nullptr) {
+        sprof.scheduler(toString(ck->schedulerType()));
+      }
+
+      // Get device index from the first input tensor if available
+      int8_t device_index = 0;
+      if (num_inputs > 0 && input_tensors[0] != nullptr) {
+        device_index = static_cast<int8_t>(input_tensors[0]->device().index());
+      } else if (num_outputs > 0 && output_tensors[0] != nullptr) {
+        device_index = static_cast<int8_t>(output_tensors[0]->device().index());
+      }
+
+      FusionProfiler::segment(group_id).setDevice(device_index);
+      sprof.startKernel();
+    }
+
+    CompiledKernel* compiled_kernel = launch_kernel_ptr->compiledKernel();
+
+    NVF_CHECK(
+        compiled_kernel != nullptr,
+        "CompiledKernel pointer is null for group_id ",
+        group_id);
+    NVF_CHECK(
+        compiled_kernel->cudaExecutable() != nullptr,
+        "CUDA executable is null for group_id ",
+        group_id);
+
+    auto index_type = compiled_kernel->kernel()->indexType();
+
+    if (isDebugDumpEnabled(DebugDumpOption::IndexType)) {
+      debug() << "Index type: " << index_type << std::endl;
+    }
+
+    std::vector<std::vector<std::byte>> arg_bytes;
+    arg_bytes.reserve(num_inputs + num_outputs);
+
+    std::vector<void*> kernel_args;
+    kernel_args.reserve(num_inputs + num_outputs);
+
+    const auto& kernel_inputs = compiled_kernel->kernel()->inputs();
+    const auto& kernel_outputs = compiled_kernel->kernel()->outputs();
+
+    // Create ExpressionEvaluator and bind input tensors to it
+    // This is needed for inferAndValidateAllocationSizesAndStrides to
+    // evaluate symbolic split factors and dimension extents when
+    // transforming between logical and allocation domains
+    ExpressionEvaluator expr_eval;
+    for (int64_t i = 0; i < num_inputs; ++i) {
+      if (input_tensors[i] != nullptr &&
+          i < static_cast<int64_t>(kernel_inputs.size())) {
+        expr_eval.bind(kernel_inputs[i], PolymorphicValue(*input_tensors[i]));
+      }
+    }
+
+    // Lambda to encode a tensor with allocation strides if available
+    auto encodeTensor = [&](at::Tensor* tensor, Val* tv_val) {
+      std::vector<int64_t> sizes = tensor->sizes().vec();
+      std::vector<int64_t> strides;
+
+      // Use allocation strides if the tensor has an allocation domain
+      if (auto* tv = dynamic_cast<TensorView*>(tv_val)) {
+        if (tv->hasAllocation()) {
+          auto [alloc_sizes, alloc_strides] =
+              inferAndValidateAllocationSizesAndStrides(*tensor, tv, expr_eval);
+          strides = alloc_strides;
```

Comment:
Unfortunately we lose a lot of the gains we would like to get from this work calculating the allocation strides. I plan to address this in a follow up pr.